Update customer-managed-keys.md (#77204)
Azure/azure-content: articles/sentinel/customer-managed-keys.md

---
# Set up Azure Sentinel customer-managed key

> This article provides background information and steps to configure a customer-managed key (CMK) for Azure Sentinel. CMK enables all data stored in Azure Sentinel to be encrypted in all relevant storage resources with an Azure Key Vault key created and owned by you.

Reviewer: ```suggestion
This article provides background information and steps to configure a [customer-managed key (CMK)](../azure-monitor/logs/customer-managed-keys.md) for Azure Sentinel. CMK allows you to give all data stored in Azure Sentinel - already encrypted by Microsoft in all relevant storage resources - an extra layer of protection with an encryption key created and owned by you and stored in your [Azure Key Vault](../key-vault/general/overview.md).
```

> [!NOTE]

> > - The Azure Sentinel CMK capability is provided only to **new workspaces**.

Reviewer: ```suggestion
> - The Azure Sentinel CMK capability is provided only to **workspaces in Log Analytics dedicated clusters** that have **not already been onboarded to Azure Sentinel**.
>
> - The following CMK-related changes **are not supported**, as they will be ineffective (Azure Sentinel data will continue to be encrypted only by the Microsoft-managed key and not the customer-managed key):
>    - Enabling CMK on a workspace **already onboarded** to Azure Sentinel.
>    - Enabling CMK on a cluster containing Sentinel-onboarded workspaces.
>    - Linking a Sentinel-onboarded non-CMK workspace to a CMK-enabled cluster.
```

>

> > - Onboarding a CMK workspace to Sentinel is supported only via REST API and not via the Azure Portal. Azure Resource Manager (ARM) templates are currently not supported for CMK onboarding.

Reviewer: Move this paragraph up (before Line 25) or down (just before the "System Assigned Identities" paragraph).

>

> > - Once a workspace is onboarded as a CMK workspace, setting it as a non-CMK workspace by de-linking it from the dedicated Log Analytics CMK cluster, or setting the dedicated Log Analytics cluster as non-CMK is not supported and might lead to undefined and undesired behavior.

Reviewer: ```suggestion
> - The following CMK-related changes **are also not supported**, as they may lead to undefined and undesired behavior:
>    - Disabling CMK on a workspace already onboarded to Azure Sentinel.
>    - Setting a Sentinel-onboarded, CMK-enabled workspace as a non-CMK workspace by de-linking it from its CMK-enabled dedicated cluster.
>    - Disabling CMK on a CMK-enabled Log Analytics dedicated cluster.
```

>

> > - Once a workspace was onboarded to Azure Sentinel as a non-CMK workspace, linking it to a CMK Log Analytics cluster, or setting the Log Analytic cluster its currently linked to, as a CMK cluster, is not supported, and will lead to its Azure Sentinel data being encrypted by a Microsoft Managed Key instead of the Customer Managed Key.

Reviewer: ```suggestion
```
(This paragraph was replaced by adding its contents to Line 27 in my second comment above.)

>

> > - Azure Sentinel supports System Assigned Identities in CMK configuration. Therefore, the dedicated Log Analytics cluster's identity should be of System Assigned type. It is recommended that it will be the identity automatically assigned to the Log Analytics cluster upon its creation.

Reviewer: Is this referring to "System Assigned **Managed** Identities"? If so, please amend the text (preferably, include link to its documentation page).

Reviewer: ```suggestion
> - Azure Sentinel supports System Assigned Identities in CMK configuration. Therefore, the dedicated Log Analytics cluster's identity should be of **System Assigned** type. It is recommended to use the identity automatically assigned to the Log Analytics cluster at the time of its creation.
```

> > - Currently, only System Assigned Identities are supported with Azure Sentinel CMK. Therefore, the dedicated Log Analytics cluster's identity should be of System Assigned type. It is recommended that it will be the identity automatically assigned to the Log Analytics cluster upon its creation.

Reviewer 2: @ehudklein  can we make this line more positive? 
like we support Assigned Identities with Azure Sentinel CMK

>

> > - Changing the Customer Managed Key to another key (with another Uri) is currently not supported. Changing the key should be done by rotating it.

Reviewer: ```suggestion
> - Changing the customer-managed key to another key (with another URI) is currently **not supported**. Changing the key should be done by [rotating it](../azure-monitor/logs/customer-managed-keys.md#key-rotation).
```

>

> > - Please consult Azure Sentinel's Product Group about any CMK changes to a production workspace or a Log Analytics cluster at azuresentinelCMK@microsoft.com.

Reviewer: ```suggestion
> - Please consult Azure Sentinel's Product Group at azuresentinelCMK@microsoft.com:
>    - Before making any CMK changes to a production workspace or a Log Analytics cluster.
>    - Upon completing the steps in this guide, and **prior to using the workspace**, for onboarding confirmation.
```

>

> > - Upon completing the steps in this guide, and prior to using the workspace, please contact azuresentinelCMK@microsoft.com for onboarding confirmation.

Reviewer: ```suggestion
```

(This paragraph was replaced by content in the previous comment.)

>

> > - The CMK capability requires a dedicated Log Analytics cluster with 1TB a day or more of reserved capacity. Several workspaces can be linked to the same dedicated cluster, and they will share the same customer managed key.

Reviewer: ```suggestion
> - The CMK capability requires a Log Analytics dedicated cluster with at least a 1 TB/day commitment tier. Several workspaces can be linked to the same dedicated cluster, and they will share the same customer-managed key.
```

>

> > - You will receive information about additional pricing when you apply to Microsoft to provision CMK on your Azure subscription. Learn more about [Log Analytics dedicated cluster](../azure-monitor/logs/manage-cost-storage.md#log-analytics-dedicated-clusters).

Reviewer: "Apply to" Microsoft? Or contact Microsoft as described in the previous step?

Author: when the customer creates the dedicated cluster in LogA

Author: I changes it and included a full path to the logA pricing for dedicated clusters

## How CMK works 

> The Azure Sentinel solution uses several storage resources for log collection and features, including a Log Analytics dedicated cluster. As part of the Azure Sentinel CMK configuration, you will have to configure the CMK settings on the related Log Analytics dedicated cluster. Data saved by Azure Sentinel in storage resources other than Log Analytics will also be encrypted using the customer managed key configured for the dedicated Log Analytics cluster.

Reviewer: ```suggestion
The Azure Sentinel solution uses several storage resources for log collection and features, including a Log Analytics dedicated cluster. As part of the Azure Sentinel CMK configuration, you will have to configure the CMK settings on the related Log Analytics dedicated cluster. Data saved by Azure Sentinel in storage resources other than Log Analytics will also be encrypted using the customer-managed key configured for the dedicated Log Analytics cluster.
```

> Learn more about [CMK](../azure-monitor/logs/customer-managed-keys.md#customer-managed-key-overview).

Reviewer: ```suggestion
See the following additional relevant documentation:
- [Azure Monitor customer-managed keys (CMK)](../azure-monitor/logs/customer-managed-keys.md).
- [Azure Key Vault](../key-vault/general/overview.md).
- [Log Analytics dedicated clusters](../azure-monitor/logs/logs-dedicated-clusters.md).
```

To provision CMK, follow these steps: 

1.  Create an Azure Key Vault and generate or import a key.

2.  Enable CMK on your Log Analytics workspace.

3.  Register to the Cosmos DB Resource Provider.

4.  Add an access policy to your Azure Key Vault instance.

5.  Onboard the workspace to Azure Sentinel via the onboarding API.

> ### STEP 1: Create an Azure Key Vault and storing key

Reviewer: ```suggestion
### STEP 1: Create an Azure Key Vault and generate or import a key
```

Onboard the workspace to Azure Sentinel via the Onboarding API. For more information, see [Azure Sentinel Onboarding](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/OnboardingStates.json).

## Key Encryption Key revocation or deletion

In the event that a user revokes the key encryption key, either by deleting it or removing access for the dedicated cluster and Cosmos DB Resource Provider, Azure Sentinel will
honor the change and behave as if the data is no longer available, within one hour. At this point, any operation that uses persistent storage resources such as
data ingestion, persistent configuration changes, and incident creation, will be prevented. Previously stored data will not be deleted but will remain
inaccessible. Inaccessible data is governed by the data-retention policy and will be purged in accordance with that policy.

The only operation possible after the encryption key is revoked or deleted is account deletion.

If access is restored after revocation, Azure Sentinel will restore access to the data within an hour.

> The way to revoke access is by disabling the customer managed key in the key vault, or deleting the access policy to the key, for both the dedicated Log Analytics cluster and Cosmos DB. Revoking access by removing the key from the dedicated Log Analytics cluster, or by removing the identity associated with the dedicated Log Analytics cluster is not supported.

Reviewer 2: maybe - "revoking access is by ..." ? or "In order to revoke access ..."?

> Access to the data can be revoked by disabling the customer managed key in the key vault, or deleting the access policy to the key, for both the dedicated Log Analytics cluster and Cosmos DB. Revoking access by removing the key from the dedicated Log Analytics cluster, or by removing the identity associated with the dedicated Log Analytics cluster is not supported.

Reviewer: ```suggestion
Access to the data can be revoked by disabling the customer-managed key in the key vault, or deleting the access policy to the key, for both the dedicated Log Analytics cluster and Cosmos DB. Revoking access by removing the key from the dedicated Log Analytics cluster, or by removing the identity associated with the dedicated Log Analytics cluster is not supported.
```

To understand more about how this works in Azure Monitor, see [Azure Monitor CMK revocation](../azure-monitor/logs/customer-managed-keys.md#key-revocation).

> ## Customer managed key rotation

Reviewer: ```suggestion
## Customer-managed key rotation
```

You can disable the previous version of the key after 24 hours, or after the Azure Key Vault audit logs no longer show any activity that uses the previous
version.

After rotating a key, you must explicitly update the dedicated Log Analytics cluster resource in Log
Analytics with the new Azure Key Vault key version. For more information, see [Azure Monitor CMK rotation](../azure-monitor/logs/customer-managed-keys.md#key-rotation).

> ## Replacing a customer managed key

Reviewer: ```suggestion
## Replacing a customer-managed key
```

> Azure Sentinel and Log Analytics support replacing a customer managed key. In order to replace the key, create another key, in the same key vault or another key vault, and configure it according to the key creation instructions above. Then, update the dedicated Log Analytics cluster with the new key. Sentinel will detect the key change and will use it across all Azure Sentinel's data storage resources within 1 hour.

Reviewer: ```suggestion
Azure Sentinel and Log Analytics support replacing a customer-managed key. In order to replace the key, create another key, either in the same key vault or in another key vault, and configure it according to the key creation instructions above. Then, update the dedicated Log Analytics cluster with the new key. Sentinel will detect the key change and will use it across all Azure Sentinel's data storage resources within one hour.
```
